### PR TITLE
🐛 Only watch GCPCluster / reconcile backend services on Machines

### LINKED ISSUE #167: Latest master doesn't reconcile GCP clusters
/kind bug

**What steps did you take and what happened:**

The latest master code doesn't seem to create the cluster controller. I can see in the logs that the machine controller gets started up twice: 
```
I0919 15:19:35.790213       1 controller.go:134] controller-runtime/controller "level"=0 "msg"="Starting Controller"  "controller"="gcpmachine"
I0919 15:19:35.790227       1 controller.go:134] controller-runtime/controller "level"=0 "msg"="Starting Controller"  "controller"="gcpmachine"
I0919 15:19:35.890373       1 controller.go:154] controller-runtime/controller "level"=0 "msg"="Starting workers"  "controller"="gcpmachine" "worker count"=10
I0919 15:19:35.890374       1 controller.go:154] controller-runtime/controller "level"=0 "msg"="Starting workers"  "controller"="gcpmachine" "worker count"=10
```
I believe this is because of [this](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/166/files#diff-89b0d50aa8ae6be0b19f5d317506bfb4L51) line that got changed in #166 specifically.

Changing that block of code to this seems to work as expected, with both controllers getting created.
```
	return ctrl.NewControllerManagedBy(mgr).
		WithOptions(options).
		For(&infrav1.GCPCluster{}).
		Complete(r) 
```

```
I0919 15:21:55.740503       1 controller.go:134] controller-runtime/controller "level"=0 "msg"="Starting Controller"  "controller"="gcpcluster"
I0919 15:21:55.740503       1 controller.go:134] controller-runtime/controller "level"=0 "msg"="Starting Controller"  "controller"="gcpmachine"
I0919 15:21:55.840679       1 controller.go:154] controller-runtime/controller "level"=0 "msg"="Starting workers"  "controller"="gcpcluster" "worker count"=10
I0919 15:21:55.840679       1 controller.go:154] controller-runtime/controller "level"=0 "msg"="Starting workers"  "controller"="gcpmachine" "worker count"=10
```

I'm not sure if that's actually the correct configuration above though. If I can get some feedback that that's correct, I'll happily submit a PR with the fix.

**What did you expect to happen:**

Newly created cluster gets reconciled.

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vincepri@vmware.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #167

### CODE REVIEW

**Comment by rsmitty** [comment]
Don't think I can /lgtm unfortunately. But yeah, LGTM.

**Comment by dims** [comment]
@vincepri wanna move the go.* change to another PR?

File: go.sum [hunk 1]
```diff
@@ -267,8 +267,8 @@ k8s.io/kube-openapi v0.0.0-20180731170545-e3762e86a74c/go.mod h1:BXM9ceUBTj2QnfH
 k8s.io/utils v0.0.0-20190506122338-8fab8cb257d5/go.mod h1:sZAwmy6armz5eXlNoLmJcl4F1QuKu7sr+mFQ0byX7Ew=
 k8s.io/utils v0.0.0-20190809000727-6c36bc71fc4a h1:uy5HAgt4Ha5rEMbhZA+aM1j2cq5LmR6LQ71EYC2sVH4=
 k8s.io/utils v0.0.0-20190809000727-6c36bc71fc4a/go.mod h1:sZAwmy6armz5eXlNoLmJcl4F1QuKu7sr+mFQ0byX7Ew=
-sigs.k8s.io/cluster-api v0.2.2-0.20190911143525-1202244af3fe h1:Be8GBTi8u9fUFf1x2F+OwLtVIqXaQfU55hmtKGR8R8k=
-sigs.k8s.io/cluster-api v0.2.2-0.20190911143525-1202244af3fe/go.mod h1:OE3zqLCNa1NclHYg6L1cidvNh0UHQak7kLLyBGxFGos=
+sigs.k8s.io/cluster-api v0.2.3-0.20190918140333-44491e485ce2 h1:94JO9gtfStBBv2RDdBnwFhYxsEOQ9VGbj5DV9PDhH8g=
```
Comment (dims) [inline go.sum]: didn't i rev this up yesterday? (also unrelated to the PR, can we please move it out)

**Comment by dims** [comment]
/lgtm

sorry my bad, i had touched just go.mod yesterday. so let's get this fixed

**Comment by jdef** [comment]
I pulled in this patch and now I see:
```
E0919 18:14:35.747243       1 controller.go:218] controller-runtime/controller "msg"="Reconciler error" "error"="failed to reconcile LB attachment: googleapi: Error 400: Invalid value for field 'resource.backends[0].group': 'test1-apiserver-us-central1-a'. The URL is malformed., invalid\nfailed to update backend service\nsigs.k8s.io/cluster-api-provider-gcp/cloud/services/compute.(*Service).UpdateBackendServices\n\t/workspace/cloud/services/compute/loadbalancers.go:172\nsigs.k8s.io/cluster-api-provider-gcp/controllers.(*GCPMachineReconciler).reconcileLBAttachment\n\t/workspace/controllers/gcpmachine_controller.go:323\nsigs.k8s.io/cluster-api-provider-gcp/controllers.(*GCPMachineReconciler).reconcile\n\t/workspace/controllers/gcpmachine_controller.go:232\nsigs.k8s.io/cluster-api-provider-gcp/controllers.(*GCPMachineReconciler).Reconcile\n\t/workspace/controllers/gcpmachine_controller.go:162\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:216\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:192\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker\n\t/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0/pkg/internal/controller/controller.go:171\nk8s.io/apimachinery/pkg/util/wait.JitterUntil.func1\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190704094733-8f6ac2502e51/pkg/util/wait/wait.go:152\nk8s.io/apimachinery/pkg/util/wait.JitterUntil\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190704094733-8f6ac2502e51/pkg/util/wait/wait.go:153\nk8s.io/apimachinery/pkg/util/wait.Until\n\t/go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190704094733-8f6ac2502e51/pkg/util/wait/wait.go:88\nruntime.goexit\n\t/usr/lib/golang/src/runtime/asm_amd64.s:1337"  "controller"="gcpmachine" "request"={"Namespace":"default","Name":"test1-controlplane-0"}
```

**Pull request merged**